Source/WebCore/accessibility/mac/WebAccessibilityObjectWrapperMac.mm

```diff
@@ -3556,6 +3510,24 @@ - (id)accessibilityAttributeValue:(NSString*)attribute forParameter:(id)paramete
         return parent ? [NSValue valueWithRect:parent->convertFrameToSpace(FloatRect(rect), AccessibilityConversionSpace::Page)] : nil;
     }
 
+    if ([attribute isEqualToString:NSAccessibilityTextMarkerDebugDescriptionAttribute])
+        return AXTextMarker { textMarker }.debugDescription();
+
+    if ([attribute isEqualToString:NSAccessibilityTextMarkerRangeDebugDescriptionAttribute])
+        return AXTextMarkerRange { textMarkerRange }.debugDescription();
+
+#if ENABLE(TREE_DEBUGGING)
+    if ([attribute isEqualToString:AXTextMarkerNodeDebugDescriptionAttribute]) {
+        [self showNodeForTextMarker:textMarker];
+        return nil;
+    }
+
+    if ([attribute isEqualToString:AXTextMarkerNodeTreeDebugDescriptionAttribute]) {
+        [self showNodeTreeForTextMarker:textMarker];
+        return nil;
+    }
+#endif
```

Comment:
For larger compiler-pragma blocks, I've found it nice to comment `#endif`s with the condition they are closing, i.e.:

`#endif // ENABLE(TREE_DEBUGGING)`